src/nlp/search.py

```diff
@@ -0,0 +1,366 @@
+                `index_name` (Optional `str`): The elasticsearch index name used to create the index.
+                `column` (Optional `str`): In the case of a `nlp.Dataset` input, which column to index.
+        """
+        self._indexes[column] = SparseIndex(es_client, index_name)
+        self._indexes[column].add_texts(texts, column=column)
+
+    def drop_index(self, column: str):
+        """ Drop the index with the specified column.
+
+            Args:
+                `column` (`str`): The column/identifier of the index.
+        """
+        del self._indexes[column]
+
+    def search(self, column: str, query, k: int = 10) -> Tuple[List[float], List[int]]:
+        """ Find the nearest examples indices in the dataset to the query.
+
+            Args:
+                `column` (`str`): The name/identifier of the index.
+                `query` (`Union[str, np.ndarray]`): The query as a string if `column` is a text index or as a numpy array if `column` is a vector index.
+                `k` (`int`): The number of examples to retrieve.
+
+            Ouput:
+                `scores` (`List[List[float]`): The retrieval scores of the retrieved examples.
+                `indices` (`List[List[int]]`): The indices of the retrieved examples.
+        """
+        self._check_index_is_initialized(column)
+        return self._indexes[column].search(query, k)
+
+    def search_batch(self, column: str, queries, k: int = 10) -> Tuple[List[List[float]], List[List[int]]]:
+        """ Find the nearest examples indices in the dataset to the query.
+
+            Args:
+                `column` (`str`): The column/identifier of the index.
+                `queries` (`Union[List[str], np.ndarray]`): The queries as a list of strings if `column` is a text index or as a numpy array if `column` is a vector index.
+                `k` (`int`): The number of examples to retrieve per query.
+
+            Ouput:
+                `scores` (`List[List[float]`): The retrieval scores of the retrieved examples per query.
+                `indices` (`List[List[int]]`): The indices of the retrieved examples per query.
+        """
+        self._check_index_is_initialized(column)
+        return self._indexes[column].search_batch(queries, k)
+
+    def get_nearest(self, column: str, query, k: int = 10) -> Tuple[List[float], List[dict]]:
+        """ Find the nearest examples in the dataset to the query.
+
+            Args:
+                `column` (`str`): The column/identifier of the index.
+                `query` (`Union[str, np.ndarray]`): The query as a string if `column` is a text index or as a numpy array if `column` is a vector index.
+                `k` (`int`): The number of examples to retrieve.
+
+            Ouput:
+                `scores` (`List[List[float]`): The retrieval scores of the retrieved examples.
+                `examples` (`List[List[dict]]`): The retrieved examples.
+        """
+        self._check_index_is_initialized(column)
```

Comment:
(nit-pick) you're checking this two times here

Reply:
Better twice than none x)

Reply:
`_check_index_is_initialized ` is actually used to factorize the `MissingIndex` error more than to check twice that the index has been initialized.